pkgs/build-support/trivial-builders.nix

```diff
@@ -344,4 +344,35 @@ rec {
   # Copy a list of paths to the Nix store.
   copyPathsToStore = builtins.map copyPathToStore;
 
+
+  /* Applies a list of patches to a source directory.
+   *
+   * Examples:
+   *
+   * # Patching nixpkgs:
+   * applyPatches {
+   *   name = "nixpkgs-patched";
+   *   src = pkgs.path;
+   *   patches = [
+   *     (pkgs.fetchpatch {
+   *       url = https://github.com/NixOS/nixpkgs/commit/1f770d20550a413e508e081ddc08464e9d08ba3d.patch;
+   *       sha256 = "1nlzx171y3r3jbk0qhvnl711kmdk57jlq4na8f8bs8wz2pbffymr";
+   *     })
+   *   ];
+   * }
+   */
+  applyPatches =
+      { src
+      , name
+      , patches   ? []
+      , postPatch ? ""
+      , ...
```

Comment:
I do not think that ```...``` should be allowed, it gives ```applyPatches``` full power of ```mkDerivation``` basicaly making it an alias.
My idea for that function so it was almost always used with 2 arguments: ```src``` + ```patches``` or ```src``` + ```postPatch``` (```name``` had a default value of ```"${src.name}-patched"```) just making a patched copy of a derivation's content